src/project/solve_group.rs

```diff
@@ -56,13 +56,12 @@ impl<'p> SolveGroup<'p> {
     pub fn environments(
         &self,
     ) -> impl Iterator<Item = Environment<'p>> + DoubleEndedIterator + ExactSizeIterator + 'p {
-        self.solve_group
-            .environments
-            .iter()
-            .map(|env_idx| Environment {
-                project: self.project,
-                environment: &self.project.manifest.parsed.environments.environments[*env_idx],
-            })
+        self.solve_group.environments.iter().map(|env_idx| {
+            Environment::new(
+                self.project,
+                &self.project.manifest.parsed.environments.environments[*env_idx],
```

Comment:
I don't like the `[]` indexing too much because it can panic, I feel its better to handle this explicitly, by varing what applies in this case

* Panicking with a better error message, if it not being there is a program error
* Returning a Result
* Filtering out the environments that do not match.

Reply:
could you please suggest what would be the best scenario in this case?
I just reformatted this code so I'm unaware of solve_groups matching behaviour when environment is missing. 
Should I throw an miette: error ?

Reply:
I would leave as is. If this code panics than some other thing is very wrong.